.cascade-code/Chapman.edu/_components/left_nav/left_nav_children.vtl

```diff
@@ -0,0 +1,137 @@
+## Formerly LeftNavNoSiblings. Used for index pages.
+## New left nav format that works without a block
+
+#set ( $pagePath = $currentPage.path )
+#set ( $parentDir = $currentPage.parentFolder )
+
+## Builds the left navigation
+#macro ( buildLeftNav )
+    #buildLeftNavTitle
+    <ul>
+    #set ( $children = $parentDir.children )
+    #foreach ( $child in $children )
+        #set ( $isVisible = $child.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+
+        ## Child is a page
+        #if ( $child.assetType == 'page' && $isVisible && ($child.name != 'index') )
+            #buildChildPage($child)
+
+        ## Child is a folder
+        #elseif ( $child.assetType == 'folder' && $isVisible )
+            ## For folders, also need to check visibility of index page
+            #set ( $folderIndexPage = $_.locatePage("${child.path}/index") )
+            #set ( $visibleIndex = $folderIndexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+            ## Titles for folders are set to the name of its index page
+            #set ( $childTitle = ${_EscapeTool.xml($folderIndexPage.label)} )
+            
+            ## Only if the child folder's index page is also visible
+            #if ( $visibleIndex )
+                #set ( $visibleChildren = [] )
+                #getVisibleChildren($child.children)
```

Comment:
I see the inspiration behind the pattern here (a lack of proper functions in Velocity) but I don't like it. It obscures the fact that the var `$visibleChildren` is being used as a global.

Reply:
I can remove the macro and just put all the code back in here.

Reply:
Before doing that, I'd take a look at the bigger picture. If you restructured this macro like so:

```
#macro ( buildLeftNav )
    #buildLeftNavTitle
    <ul>
    #set ( $children = $parentDir.children )
    #foreach ( $child in $children )
      buildLeftNav($child)
```

Maybe that would help reorganize things. The logic could be contained in that macro (and that macro itself might take advantage of other macros if it makes sense and isn't getting too convoluted.)

It's true you're still going to have that block of logic somewhere. But this might help better isolate and encapsulate the logic so that it doesn't interfere with a broader view of what's happening.